packages/discovery-provider/integration_tests/tasks/test_index_payment_router.py

```diff
@@ -916,3 +923,54 @@ def test_process_payment_router_txs_details_skip_unknown_PDA_ATAs(app):
             .first()
         )
         assert transaction_record is None
+
+
+def test_process_payment_router_tx_details_access_types(app):
```

Comment:
nit: It's easier to read test output if each test contains one clearly described test case. Especially in pytest where the assertions can be a bit cryptic to read when they fail and we are reusing variables here. I'm thinking mostly about the CI case.